src/auth/src/signer/iam.rs

```diff
@@ -141,14 +139,30 @@ async fn sign_blob_call(
         }
     };
 
-    client
+    let response = client
         .post(url)
         .header("Content-Type", "application/json")
         .headers(source_headers.clone())
         .json(&body)
         .send()
         .await
-        .map_err(|e| gax::error::Error::transport(source_headers, e))
+        .map_err(gax::error::Error::io)?;
```

Comment:
Seems like this should go to a separate PR, and probably deserves a test (using `httptest` should not be that hard).